gammapy/image/models/core.py

```diff
@@ -170,6 +172,136 @@ def evaluate(lon, lat, lon_0, lat_0, r_0):
+    normalization, in units of :math:`\text{sr}^{-1}`.
+    
+    The model is defined on the celestial sphere, by computing angles and distances with the functions defined in
+    `astropy.coordinates.angle_utilities`, and setting the normalization such that:
+
+    .. math::
+       \int_{4\pi}\phi(\text{lon}, \text{lat}) \,d\Omega = 1\,.
+
+
+    Parameters
+    ----------
+    lon_0 : `~astropy.coordinates.Longitude`
+        :math:`\text{lon}_0`: `lon` coordinate for the center of the ellipse.
+    lat_0 : `~astropy.coordinates.Latitude`
+        :math:`\text{lat}_0`: `lat` coordinate for the center of the ellipse.
+    semi_major : `~astropy.coordinates.Angle`
+        :math:`a`: length of the major semiaxis, in angular units.
+    e : `float`
+        Eccentricity of the ellipse (:math:`0< e< 1`).
+    theta : `~astropy.coordinates.Angle`
+        :math:`\theta`: 
+        Rotation angle of the major semiaxis.  The
+        rotation angle increases clockwise (i.e., East of North) from the positive `lon`
+        axis.
+
+
+        Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        from gammapy.image.models.core import *
+        import astropy.units as u
+        from gammapy.maps import Map, WcsGeom
+        import matplotlib.pyplot as plt
+        from matplotlib import rc
+        rc('font',**{'family':'sans-serif','sans-serif':['Helvetica']})
+        ## for Palatino and other serif fonts use:
+        #rc('font',**{'family':'serif','serif':['Palatino']})
+        rc('text', usetex=True)
+
+        ell=SkyEllipse(2*u.deg,2*u.deg,1*u.deg,.8,30*u.deg)
+
+        m_geom = WcsGeom.create(binsz=.01,width=(3,3),skydir=(2,2),coordsys="GAL",proj="AIT")
+        coords=m_geom.get_coord()
+        lon=coords.lon*u.deg
+        lat=coords.lat*u.deg  
+        vals=ell(lon,lat)
+        mymap=Map.from_geom(m_geom,data=vals.value)
+
+        fig,ax,_=mymap.plot()
+        ax.scatter(2, 2, transform=ax.get_transform('galactic'), s=20,edgecolor='red', facecolor='red')
+        plt.text(2.08, 2.06, r'$(l_0,b_0)$' ,transform=ax.get_transform('galactic'), fontsize=12)
+        plt.plot([2,2+np.cos(np.pi/6)],[2,2+np.sin(np.pi/6)], transform=ax.get_transform('galactic'))
+        ax.hlines(y=2,color='r', linestyle='--',transform=ax.get_transform('galactic'),xmin=0,xmax=5)
+        plt.text(2.4, 2.06, r'$\theta$' ,transform=ax.get_transform('galactic'), fontsize=12)
+
+        plt.show()
+    """
+
+    def __init__(self, lon_0, lat_0, semi_major, e, theta):
+        self.parameters = Parameters(
+            [
+                Parameter("lon_0", Longitude(lon_0)),
+                Parameter("lat_0", Latitude(lat_0)),
+                Parameter("semi_major", Angle(semi_major)),
+                Parameter("e", e, min=0, max=1),
+                Parameter("theta", Angle(theta)),
+            ]
+        )
+
+    def find_foci(lon_0, lat_0, semi_major, e, theta):
```

Comment:
I think it's not super-useful to have this as a separate method, just because I don't think it will be called by users. It could be in the `.evaluate()` method as well...do as you prefer.